crates/durability/src/imp/local.rs

```diff
@@ -73,48 +84,55 @@ pub struct Local<T> {
     /// This is mainly for observability purposes, and can thus be updated with
     /// relaxed memory ordering.
     queue_depth: Arc<AtomicU64>,
-    /// Handle to the [`PersisterTask`], allowing to drain the `queue` when
-    /// explicitly dropped via [`Self::close`].
-    persister_task: JoinHandle<()>,
+    /// Channel to request the actor to exit.
+    shutdown: mpsc::Sender<ShutdownReply>,
 }
 
 impl<T: Encode + Send + Sync + 'static> Local<T> {
-    /// Create a [`Local`] instance at the `root` directory.
+    /// Create a [`Local`] instance at the `replica_dir`.
     ///
-    /// The `root` directory must already exist.
+    /// `replica_dir` must already exist.
     ///
     /// Background tasks are spawned onto the provided tokio runtime.
     ///
     /// We will send a message down the `on_new_segment` channel whenever we begin a new commitlog segment.
     /// This is used to capture a snapshot each new segment.
     pub fn open(
-        root: CommitLogDir,
+        replica_dir: ReplicaDir,
         rt: tokio::runtime::Handle,
         opts: Options,
         on_new_segment: Option<Arc<OnNewSegmentFn>>,
-    ) -> io::Result<Self> {
+    ) -> Result<Self, OpenError> {
         info!("open local durability");
 
-        let clog = Arc::new(Commitlog::open(root, opts.commitlog, on_new_segment)?);
-        let (queue, rx) = mpsc::unbounded_channel();
+        // For backwards-compatibility, we keep using the `db.lock` file.
```

Comment:
Could you extend this comment to say what we'd do if backcompat was not a concern?